Makefile

```diff
@@ -59,3 +66,73 @@ gscholar_alerts/extracted_citations.jsonl: gscholar_alerts/eml/
 
 gscholar_alerts/citations.jsonl: gscholar_alerts/extracted_citations.jsonl
 	jq -c 'select(.title != null and .authors != null) | del(.idx, .date, .data, .ref, .link)' $< >$@
+
+
+
+# HF Dataset Repos Updating
+
+# Since we dont know which years will exist in the jsonl,
+# better to use a sentinel file to track their timestamp as well.
+extract-citations.done: gscholar_alerts/citations.jsonl
+	mkdir -p $(TRACKED_FILES_BASEDIR)/citations
+	BASEDIR=$(TRACKED_FILES_BASEDIR)/citations ; \
+	YEARS=$$(cat tmp/citations.jsonl | jq -r ."year" | sort | uniq) ; \
+	for YEAR in $$YEARS; do \
+		jq -c "select(."year" == \"$$YEAR\")" tmp/citations.jsonl > "$$BASEDIR/$$YEAR.jsonl"; \
+	done
+	touch $@
+
+# TODO: Slightly inefficient now, if one .bib file has newer timestamp, redoes every year's .csv file.
```

Comment:
Even older BibTeX files might see updates: format errors, updated metadata, replacement of broken URLs.

So, the timestamp is probably the best way to rely on.